<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" >


<link
  name="imu_link">
  <inertial>
    <origin
      xyz="0 6.1424E-20 0"
      rpy="0 0 0" />
    <mass
      value="0.0070686" />
    <inertia
      ixx="4.5651E-07"
      ixy="4.6312E-37"
      ixz="-3.2701E-21"
      iyy="4.5651E-07"
      iyz="3.0522E-23"
      izz="7.9522E-07" />
  </inertial>
  <visual>
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <geometry>
      <mesh
        filename="file://$(find mmdt_ros2)/meshes/imu.STL" />
    </geometry>
    <material
      name="">
      <color
        rgba="0.79216 0.81961 0.93333 1" />
    </material>
  </visual>
  <collision>
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <geometry>
      <mesh
        filename="file://$(find mmdt_ros2)/meshes/imu.STL" />
    </geometry>
  </collision>
</link>

<joint
  name="eje_imu"
  type="fixed">
  <origin
    xyz="-0.152 0 0.052"
    rpy="0 0 0" />
  <parent
    link="chasis" />
  <child
    link="imu_link" />
  <axis
    xyz="0 0 0" />
</joint>


  <gazebo reference="imu_link">
    <gravity>true</gravity>
    <sensor name="imu_sensor" type="imu">
      <always_on>true</always_on>
      <update_rate>100</update_rate>
      <visualize>true</visualize>
      <topic>__default_topic__</topic>
      <plugin filename="libgazebo_ros_imu_sensor.so" name="imu_plugin">
        <topicName>imu</topicName>
        <bodyName>imu_link</bodyName>
        <updateRateHZ>10.0</updateRateHZ>
        <gaussianNoise>0.0</gaussianNoise>
        <xyzOffset>0 0 0</xyzOffset>
        <rpyOffset>0 0 0</rpyOffset>
        <frameName>imu_link</frameName>
        <initialOrientationAsReference>false</initialOrientationAsReference>
      </plugin>
      <pose>0 0 0 0 0 0</pose>
    </sensor>
  </gazebo>

</robot>